internal/remote/skr_context.go

```diff
@@ -237,10 +237,22 @@ func (s *SkrContext) getRemoteKyma(ctx context.Context) (*v1beta2.Kyma, error) {
 // syncWatcherLabelsAnnotations adds required labels and annotations to the skrKyma.
 // It returns true if any of the labels or annotations were changed.
 func syncWatcherLabelsAnnotations(kcpKyma, skrKyma *v1beta2.Kyma) bool {
-	labels, labelsChanged := collections.MergeMaps(skrKyma.Labels, map[string]string{
-		shared.WatchedByLabel: shared.WatchedByLabelValue,
-		shared.ManagedBy:      shared.ManagedByLabelValue,
-	})
+	labelsMap := map[string]string{}
```

Comment:
this function is used to sync watcher related label as the name suggests, let's not misuse it.

Reply:
Makes sense, extracted the logic into an independent function